lib/restbase.js

```diff
@@ -307,6 +318,17 @@ RESTBase.prototype._request = function (req) {
             self.metrics.endTiming([statName + statusClass, statName + 'ALL'], startTime);
             throw err;
         });
+        if (match.permissions
+                && Array.isArray(match.permissions)
+                && match.permissions.length > 0) {
+            reqHandlerPromise = reqHandlerPromise.then(function(res) {
```

Comment:
The API is quite slow (about 50ms min latency for a 'hello world' request), so it's important to start those requests as early as possible in order to minimize the overall request latency. You are right that we might end up checking the authentication despite some other part of the request failing, but that's fine.